src/trace-writer.js

```diff
@@ -186,10 +202,15 @@ TraceWriter.prototype.writeSpan = function(spanData) {
 TraceWriter.prototype.queueTrace_ = function(trace) {
   var that = this;
 
+  if (!that.isActive) {
+    return;
```

Comment:
This means that we will continue attempting to queue traces even when we have failed to initialize correctly. Is there a way that this bailing out could be done earlier in the process, maybe in the plugins, so that we avoid propagating context (which incurs most of the performance overhead) if we know the agent has turned off?